### DISCUSSION TITLE
Command history missing for default commands

### DISCUSSION THREAD

**Discussion by Sripadvallabh** [python-cmd2/cmd2#1387] [Q&A] [answered]
Due to some requirement, i have made a default command by overriding the default() function as shown below:
```
    # Default command to be 'run'
    def default(self, stmt):
        self.do_run(stmt.raw)
```

Because of this, most of the time users would be invoking 'run' command directly with arguments instead of calling 'run <ARGS>':

```
(cmd2) <args> 
instead of 
(cmd2) run <args>
```
The run command doesn't have argparse options so it works seamlessly.

But unfortunately, these commands don't show up in the history when i restart my application & check using 'history' command or by Up & down arrows.

May i know if this is supported?

**Comment by kmvanbrunt** [comment, marked as answer]
Call one of the following:
1. `self.onecmd_plus_hooks(f"run {stmt.raw}")`
2. `self.onecmd(f"run {stmt.raw}")`

**Reply by Sripadvallabh** [reply]
Thanks it works.

**Reply by Sripadvallabh** [reply]
It seems we run into issues when we declare macros for default commands; The macro doesn't get expanded because stmt.raw contains the macro version & not the expanded one which sits inside stmt.args

So the solution looks like:
```
self.onecmd(f"run {stmt.command} {stmt.args}")
```

Am i right Kevin?
